### PR TITLE
Minor grammatical changes

### PR DESCRIPTION
**Pull request by InteXX**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 This library provides F# style ~discriminated~ unions for C#, using a custom type `OneOf<T0, ... Tn>`. An instance of this type holds a single value, which is one of the types in its generic argument list.
 
-I can't encourage you enough to give it a try! Due to exhaustive matching DUs provide an alternative to polymorphism when you want to have a method with guaranteed behaviour-per-type (i.e. adding an abstract method on a base type, and then implementing that method in each type). It's a really powerful tool, ask any f#/Scala dev! :)
+I can't encourage you enough to give it a try! Due to exhaustive matching DUs, OneOf provides an alternative to polymorphism when you want to have a method with guaranteed behaviour-per-type (e.g. adding an abstract method on a base type, and then implementing that method in each type). It's a really powerful tool, ask any f#/Scala dev! :)
```
Comment (bartelink) [inline README.md:11]: I think the ie was correct. While it's true that OneOf delivers this facility, the point is that it is a clean way of achieving the general pattern (if you'll pardon the pun) of pattern matching based on the DU type making you handle each case. I am not saying my wording is better than the original; just saying that the first wording was not wrong IMO, and I'm not sure that your suggesiton is better on all counts; perhaps Hayyy will find something useful across the two sets of sugestions ;)
```suggestion
I can't encourage you enough to give it a try! DUs, with the 'exhaustive matching' mechanism if affords, offers an alternative to polymorphism for defining behaviour-per-type (i.e., adding an abstract method on a base type and then implementing that method in each type). It's a really powerful tool, ask any F#/Scala dev! :)
```

Comment (InteXX) [reply README.md:11]: I see where I went wrong now. I was misreading this as an incomplete sentence:

> Due to exhaustive matching DUs provide an alternative to polymorphism ...

When in fact it's not.

But it'll be easier to visually parse if a comma is introduced:

> Due to exhaustive matching, DUs provide an alternative to polymorphism ...

As far as the `i.e.`, that bit of Latin translates to "[in other words](https://en.wikipedia.org/wiki/List_of_Latin_phrases_(I)#id_est)." I could be mistaken, but I believe the author's intent was to say "[for example](https://en.wikipedia.org/wiki/List_of_Latin_phrases_(E)#e.g.)," which is `e.g.`.

If you agree with these updates, I'll close this PR and start a new one.

Comment (bartelink) [reply README.md:11]: Definitely agree there's a missing comma or two, and that would fix the bulk of the things

I disagree that the author intended e.g. -- I believe he was trying to allude to a very clear single equivalent for people that don't know what 'exhaustive matching' or 'DUs' even are. i.e. saying "the equivalent OO idiom is to have a type hierarchy, add an abstract method, and then have the compiler force you to go and implement it at each call site". As opposed to saying something looser akin to "you know, like how the visitor pattern, abstract methods and other such techniques let you coordinate doing something in a type-directed way"

Not sure another PR is necessary - just adjust your suggestion here so the context is preserved.

Ultimately you're onto something in that it's hard to parse and not 100% clear who the wording is primarily intended to speak to as it stands -- the trick/problem is adjusting it to convey the goal to any or all of:
- people scanning really quickly
- people for whom exhaustive pattern matching is already a thing
- people who've mainly been exposed to OO terms
(At the time I scanned it I was in the middle camp, so the lack of commas did not trip me up)

Comment (InteXX) [reply README.md:11]: Fair 'nuff. I'll retract my `e.g.` and sentence structure change, and then simply place a comma between the words matching and DUs.

I'm not seeing where to adjust my commit here in this PR, though. I'm pretty sure I have to start a new one. Am I missing something?

Comment (bartelink) [reply README.md:11]: Select the line
image
click the +- icon
image
then you can suggest to yourself
and then apply it

(I'd take the time to learn to clone the repo and edit though)

or use the github.dev editor via `.`:  https://docs.github.com/en/codespaces/the-githubdev-web-based-editor#opening-the-githubdev-editor

or press `,` to create a codespace

**Comment by InteXX** [comment]
Closing to resubmit with suggested alterations.
